File: LOO 1°Parcial/Complex/Complex.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package ejemplosalumnos;
public class Complex {
    private double real, imag;
     public Complex(){
     this.real = 0;
    this.imag = 0;
     }
     public Complex(double r, double i){
     this.real = r;
    this.imag = i;
     }
     public double module(){
     return Math.sqrt(this.real * this.real + this.imag* this.imag);
     }
     public Complex conjugate(){
     Complex nuevo = new Complex(this.real, -this.imag);
    return nuevo;
     }
     public void add(Complex n1, Complex n2){
     this.real = n1.getReal() + n2.getReal();
    this.imag = n1.getImag() + n2.getImag();
     }
     public void subs(Complex n1, Complex n2){
     this.real = n1.getReal() - n2.getReal();
    this.imag = n1.getImag() - n2.getImag();
     }
     public void mult(Complex n1, Complex n2){
     this.real = n1.getReal() * n2.getReal() - n1.getImag() * n2.getImag();
    this.imag = n1.getReal() * n2.getImag() + n1.getImag() * n2.getReal();
     }
     public void divi(Complex n1, Complex n2){
     this.real = (n1.getReal() * n2.getReal() + n1.getImag() * n2.getImag()) / (n2.getReal()* n2.getReal() + n2.getImag() * n2.getImag());
    this.imag = (-n1.getReal() * n2.getImag() + n2.getReal() * n1.getImag()) / (n2.getReal() * n2.getReal() + n2.getImag() * n2.getImag());
     }
     @Override
     public String toString(){
     String str;
    str = (this.real) + (this.imag<0?"":"+") + (this.imag) + " i";
    return str;
     }
     //métodos set y get
     public void setReal(double r){
     this.real = r;
     }
     public double getReal(){
     return this.real;
     }
     public void setImag(double i){
     this.imag = i;
     }
     public double getImag() {
     return this.imag;
     }
}

File: LOO 2°Parcial/ComplejosCuaterniones/ComplejosCuaterniones/Complex.cpp
#include "Complex.h"
#include <cmath>

Complex::Complex() {
	this->real = 0;
	this->imag = 0;
}
Complex::Complex(float r, float i) {
	this->real = r;
	this->imag = i;
}
float Complex::module(void) {
	return sqrt(this->real * this->real + this->imag * this->imag);
}
Complex Complex::conjugate(void) {
	Complex nuevo(this->real, -this->imag);
	return nuevo;
}
void Complex::add(Complex n1, Complex n2) {
	this->real = n1.getReal() + n2.getReal();
	this->imag = n1.getImag() + n2.getImag();
}
void Complex::subs(Complex n1, Complex n2) {
	this->real = n1.getReal() - n2.getReal();
	this->imag = n1.getImag() - n2.getImag();
}
void Complex::mult(Complex n1, Complex n2) {
	//(a*c – b*d) + (a*d + b*c)
	this->real = n1.getReal() * n2.getReal() - n1.getImag() * n2.getImag();
	this->imag = n1.getReal() * n2.getImag() + n1.getImag() * n2.getReal();
}
void Complex::divi(Complex n1, Complex n2) {
	this->real = (n1.getReal() * n2.getReal() + n1.getImag() * n2.getImag()) / (n2.getReal() * n2.getReal() + n2.getImag() * n2.getImag());
	this->imag = (-n1.getReal() * n2.getImag() + n2.getReal() * n1.getImag()) / (n2.getReal() * n2.getReal() + n2.getImag() * n2.getImag());
}
string Complex::toString(void) {
	string str;
	str = to_string(this->real) + (this->imag < 0 ? "" : "+") + to_string(this->imag) + " i";
	return str;
}
//métodos set y get
void Complex::setReal(float r) {
	this->real = r;
}
float Complex::getReal(void) {
	return this->real;
}
void Complex::setImag(float i) {
	this->imag = i;
}
float Complex::getImag(void) {
	return this->imag;
}
